h/views/badge.py

```diff
@@ -22,6 +24,17 @@ def record_metrics(count,
     record_metric('Custom/Badge/badgeCountIsZero', int(count == 0))
 
 
+def _has_uri_ever_been_annotated(db, uri):
+    """Return `True` if a given URI has ever been annotated."""
+
+    # This check is written with SQL directly to guarantee an efficient
+    # query and minimize SQLAlchemy overhead.
+    db_conn = db.connection()
+    query = text('SELECT EXISTS(SELECT 1 FROM document_uri WHERE uri_normalized = :uri)')
+    result = db_conn.execute(query, uri=normalize(uri)).first()
+    return result[0] is True
```

Comment:
Would it be better to use `request.db.execute()` here instead of (what looks like it might be) opening a new connection?

Reply:
My reading of the docs suggests that both should be pretty much the same but I've changed it to `request.db.execute()` as that is one less line of code.